src/Button/ButtonBase.tsx

```diff
@@ -22,21 +28,27 @@ const variants = variant({
   }
 })
 
-const ButtonBase = styled.button.attrs(({disabled, onClick}) => ({
+type StyledButtonBaseProps = {
+  as?: 'button' | 'a' | 'summary' | 'input' | string | React.ReactType
+  fontSize?: FontSizeProps['fontSize']
+  variant?: 'small' | 'medium' | 'large'
+} & SystemCommonProps &
+  SystemLayoutProps &
+  SxProp
```

Comment:
Just curious, is there a reason you didn't write it this way?

```suggestion
type StyledButtonBaseProps = {
  as?: 'button' | 'a' | 'summary' | 'input' | string | React.ReactType
  variant?: 'small' | 'medium' | 'large'
} & FontSizeProps &
  SystemCommonProps &
  SystemLayoutProps &
  SxProp
```

Reply:
Hmm, also isn't the type incorrect because `ButtonBase` doesn't actually accept common, layout, and fontsize props?

Reply:
> Just curious, is there a reason you didn't write it this way?

It’s possible I thought (incorrectly) that `FontSizeProps` includes keys not used by `ButtonBase` (i.e. in addition to `'fontSize'`)—but it’s more likely the intersection simply didn’t occur to me 😅

> …[I]sn't the type incorrect because ButtonBase doesn't actually accept common, layout, and fontsize props?

I’d defined it this way to avoid needing to add those to custom types for `Button`, `ButtonDanger`, `ButtonInvisible`, `ButtonOutline` and `ButtonPrimary` (which all use `fontSize`, `COMMON`, `LAYOUT`, and `sx`), in order to make `ButtonBaseProps` more useful as a base. It didn’t occur to me that—in doing so—I’d made `ButtonBase`’s prop type itself wrong, especially because `ButtonBase` is never (that I know of) rendered outside one of those components.

It definitely makes sense to clean that up (i.e. to type `ButtonBase`’s props correctly). Thanks for catching this! 👍

Reply:
> It definitely makes sense to clean that up…

Note: @colebemis gave a good suggestion for how to do this DRYly in https://github.com/primer/components/pull/1017#discussion_r571259317

Reply:
Sidebar: Is it true that “`ButtonBase` doesn't actually accept [a] fontsize prop”? The [`propTypes` in the current published version](https://github.com/primer/components/blob/b894ead466822e42b8f659f37a37ec3c71876506/src/Button/ButtonBase.js#L43) includes `fontSize`.

Reply:
Fixed in https://github.com/primer/components/pull/1017/commits/7f8997e9748aaadd949c420b46915396508e8532